src/figuresClasses.ts

```diff
@@ -1,11 +1,103 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: 'triangle';
+
+  public color: 'red' | 'green' | 'blue';
+
+  public a: number;
+
+  public b: number;
+
+  public c: number;
+
+  constructor(
+    // eslint-disable-next-line @typescript-eslint/default-param-last
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Sides must be heigher then 0');
+    }
+
+    if (b + c <= a || a + c <= b || a + b <= c) {
+      throw new Error(
+        'The longest side must be less than the sum of the other two sides',
+      );
+    }
+
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
When you declare a constructor parameter with an access modifier TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property
```suggestion
```